src/lower.c

```diff
@@ -111,7 +111,12 @@ typedef struct {
    bool        is_const;
 } map_signal_param_t;
 
-typedef void (*lower_field_fn_t)(type_t, vcode_reg_t, vcode_reg_t, void *);
+typedef struct {
+   ident_t     hier;
+   unsigned    flags;
+} cov_toggle_ctx_t;
```

Comment:
You don't need to pass the flags through, you can just check if `tree_kind(where) == T_SIGNAL_DECL` or `T_PORT_DECL`.

Reply:
Good, point, I wasn't sure whether the nesting through elements of arrays will not change the tree kind.